databroker/tests/test_broker.py

```diff
@@ -522,8 +522,8 @@ def myplan(dets):
 
         Meant for test only.
         '''
-        yield from trigger_and_read([dets[0]], name='primary')
-        yield from trigger_and_read([dets[1]], name='secondary')
+        yield trigger_and_read([dets[0]], name='primary')
```

Comment:
Oh, I see what went wrong here. The Python 2 version of `yield from` is:

```py
for msg in trigger_and_read([dets[0]], name='primary')
    yield msg
```

Doing `yield trigger_and_read([dets[0]], name='primary')` yields the generator instance itself, not the messages can issue.

Reply:
thanks!